content/en/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Migrating telemetry and security agents from dockershim
+content_type: task 
+reviewers:
+- SergeyKanzhelev
+- jrcamp # for SignalFx (Splunk)
+weight: 70
+---
+
+<!-- overview -->
+
+With kubernetes 1.20 dockershim was deprecated. From the [Dockershim Deprecation
+FAQ](https://kubernetes.io/blog/2020/12/02/dockershim-faq/) you might already
+know that most apps do not have a direct dependency on runtime hosting
+containers. However, there are still a lot of telemetry and security agents
+that has a dependency on docker to collect containers metadata, logs and
+metrics. This document aggregates information on how to detect these
+dependencies and links on how to migrate these agents to use generic tools or
+alternative runtimes.
+
+## Telemetry and security agents
+
+There are a few ways agents may run on kubernetes cluster. Agents may run on
+nodes directly or as DaemonSets.
+
+### Why do telemetry agents talk to Docker?
+
+Historically, kubernetes was built on top of Docker. Kubernetes is managing networking and scheduling, Docker was placing and operating containers on a node. So you can get scheduling-related metadata like a pod name from Kubernetes and containers state information from Docker. Over time more runtimes were created to manage containers. Also there are projects and kubernetes features that generalize container status information extraction across many runtimes.
+
+Some agents are tied specifically to the Docker tool. The agents may run commands like [`docker
+ ps`](https://docs.docker.com/engine/reference/commandline/ps/) or [`docker
+top](https://docs.docker.com/engine/reference/commandline/top/) to list
+containers and processes or [docker
+logs](https://docs.docker.com/engine/reference/commandline/logs/) to subscribe
```

Comment:
The wrapping of lines here are terrible. It would be desirable to have each link stay on the same line, no matter how long that line may be.

There is a missing backtiq for the docker top command.

Reply:
> The wrapping of lines here are terrible.

agree. This extension I use always doing it (https://github.com/stkb/Rewrap/issues/112). 

Fixed the backtiq